paasta_tools/autoscaling/cluster_boost.py

```diff
@@ -0,0 +1,160 @@
+"""
+import logging
+from datetime import datetime
+
+from paasta_tools.utils import ZookeeperPool
+
+DEFAULT_BOOST_FACTOR = 1.5
+DEFAULT_BOOST_DURATION = 40
+
+MIN_BOOST_FACTOR = 1.0
+MAX_BOOST_FACTOR = 3.0
+
+MAX_BOOST_DURATION = 240
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class BoostAlreadyActiveError(Exception):
+    pass
+
+
+def get_zk_boost_path(region: str, pool: str) -> str:
+    return '/paasta_cluster_autoscaler/{}/{}/boost'.format(region, pool)
+
+
+def get_boosted_load(region: str, pool: str, current_load: float) -> float:
+    """Return the load to use for autoscaling calculations, taking into
+    account the computed boost, if any.
+    """
+    try:
+        zk_boost_path = get_zk_boost_path(region, pool)
+        current_time = int(datetime.now().strftime('%s'))
+
+        with ZookeeperPool() as zk:
+            end_time, boost_factor, expected_load = get_boost_values(region, pool, zk)
+
+            if current_time >= end_time:
+                # Boost period is over. If expected_load is still nozero, reset it to 0
+                if expected_load > 0:
+                    zk.set(zk_boost_path + '/expected_load', '0'.encode('utf-8'))
+
+                return current_load
+
+            # Boost is active. If expected load wasn't already computed, set it now.
+            if expected_load == 0:
+                expected_load = current_load * boost_factor
+
+                log.debug('Activating boost, storing expected load: {} in ZooKeeper'.format(expected_load))
+
+                zk.ensure_path(zk_boost_path + '/expected_load')
+                zk.set(zk_boost_path + '/expected_load', str(expected_load).encode('utf-8'))
+
+            # We return the boosted expected_load, but only if the current load isn't greater.
+            return expected_load if expected_load > current_load else current_load
+
+    except Exception as e:
+        # Fail gracefully in the face of ANY error
+        log.error('get_boost failed with: {}'.format(e))
+        return current_load
+
+
+def is_boost_active(
+    region: str,
+    pool: str,
+    zk: ZookeeperPool,
+) -> bool:
+    current_time = int(datetime.now().strftime('%s'))
+    end_time, _, _ = get_boost_values(region, pool, zk)
+    return current_time < end_time
+
+
+def get_boost_values(
+    region: str,
+    pool: str,
+    zk: ZookeeperPool,
+) -> (int, float, float):
+    # Default values, non-boost.
+    end_time = 0
+    boost_factor = 1.0
+    expected_load = 0
+
+    try:
+        zk_boost_path = get_zk_boost_path(region, pool)
+        end_time = int(zk.get(zk_boost_path + '/end_time')[0].decode('utf-8'))
+        boost_factor = float(zk.get(zk_boost_path + '/factor')[0].decode('utf-8'))
+        expected_load = float(zk.get(zk_boost_path + '/expected_load')[0].decode('utf-8'))
+
+    except Exception as e:
+        log.info(e)
+        log.debug('No boost data in Zookeeper for pool {} in region {}.'.format(
```

Comment:
please except this on exactly the missing zoned, and then just return. But don't log, this will be happening all the time, and if there is no boost then there is no boost.